packages/next/next-server/server/render.tsx

```diff
@@ -442,9 +445,20 @@ export async function renderToHTML(
     })
 
     if (isSpr && !isFallback) {
-      const data = await unstable_getStaticProps!({
-        params: isDynamicRoute(pathname) ? (query as any) : undefined,
-      })
+      const previewData = tryGetPreviewData(req, res, previewProps)
+      const data = await unstable_getStaticProps!(
```

Comment:
The type for `Unstable_getStaticProps` should be updated with `preview?: boolean, previewData?: {[key: string]: string}` in `load-components`, interesting how this breaks TypeScript detecting it. Might be worth not using `Object.assign` so it can be type checked 🤔

Reply:
Removed `Object.assign` usage.